lib/service/storage/elasticsearch.ts

```diff
@@ -1007,20 +1128,28 @@ class ElasticSearch extends Service {
    * @returns {Promise.<{ documents, total, deleted, failures: Array<{ _shardId, reason }> }>}
    */
   async deleteByQuery(
-    index,
-    collection,
-    query,
-    { refresh, size = 1000, fetch = true } = {},
+    index: string,
+    collection: string,
+    query: JSONObject,
+    {
+      refresh,
+      size = 1000,
+      fetch = true,
+    }: {
+      refresh?: boolean | "wait_for";
+      size?: number;
+      fetch?: boolean;
+    } = {},
```

Comment:
```suggestion
    {
      refresh,
      size = 1000,
      fetch = true,
    }: Pick<KRequestParams, "refresh"> & {
      size?: number;
      fetch?: boolean;
    } = {},
```

With new `KRequestParams` type we can compose type like it, to DRY